autode/opt/optimisers/robust.py

```diff
@@ -0,0 +1,458 @@
+"""
+Robust geometry optimisers, uses features from
+multiple optimisation methods. (Similar to those
+implemented in common QM packages)
+
+Only minimisers, these are not TS search/constrained optimiser
+"""
+import numpy as np
+
+from autode.opt.coordinates import CartesianCoordinates, DIC
+from autode.opt.coordinates.primitives import Distance
+from autode.opt.optimisers import CRFOptimiser, NDOptimiser
+from autode.opt.optimisers.hessian_update import FlowchartUpdate
+from autode.exceptions import OptimiserStepError
+from autode.values import GradientRMS, PotentialEnergy
+from autode.log import logger
+from itertools import combinations
+
+
+class HybridTRIMOptimiser(CRFOptimiser):
+    """
+    A hybrid of RFO and TRIM/QA optimiser, with dynamic trust
+    radius. If the RFO step is smaller in magnitude than trust radius,
+    then the step-length is controlled by using the Trust-Radius Image
+    Minimisation method (TRIM), also known as Quadratic Approximation
+    (QA). It falls-back on simple scalar scaling of RFO step, if the
+    TRIM/QA method fails to converge to the required step-size. The
+    trust radius is updated by a modification of Fletcher's method.
+
+    See References:
+
+    [1] P. Culot et. al, Theor. Chim. Acta, 82, 1992, 189-205
+
+    [2] T. Helgaker, Chem. Phys. Lett., 182(5), 1991, 503-510
+
+    [3] J. T. Golab et al., Chem. Phys., 78, 1983, 175-199
+    """
+
+    def __init__(
+        self,
+        coord_type: str = "dic",
+        init_trust: float = 0.1,
```

Comment:
@t-young31 The problem is that it clashes with `autode.opt.coordinates.primitives.Distance`. I will open an issue so that it can be fixed in a latter PR (not to clutter this one)